src/Parsers/ExpressionListParsers.cpp

```diff
@@ -830,8 +830,8 @@ class LayerWithSeparator : public Layer
 class FunctionLayer : public Layer
 {
 public:
-    explicit FunctionLayer(String function_name_, bool allow_function_parameters_ = true)
-        : function_name(function_name_), allow_function_parameters(allow_function_parameters_){}
+    explicit FunctionLayer(String function_name_, bool allow_function_parameters_ = true, bool has_database_name_ = false)
```

Comment:
as discussed, we can avoid adding `has_database_name` if pass `ASTIdentifier` for function name.

Reply:
as discussed, I tried to replace the FunctionLayer input parameters to ASTIdentifier, but this too needs updating identifier name to lowercase in some places ( for functions like "grouping"), would be better to take this up in the refactor ASTFunction name PR.